pool/object.go

```diff
@@ -145,7 +145,7 @@ func (p *objectPool) Put(obj interface{}) {
 }
 
 func (p *objectPool) trySetGauges() {
-	if time.Now().UnixNano()%sampleObjectPoolLengthEvery == 0 {
+	if atomic.AddInt32(&p.dice, 1)%sampleObjectPoolLengthEvery == 0 {
```

Comment:
mind adding a quick benchmark repro-ing the issue - will make it easier to test future changes.

Reply:
Sounds good